### PR TITLE
added functionality to tabs and draft apply course and add course button

### PR DESCRIPTION
**Pull request by huiminzhang0111**
loom explanation video: https://www.loom.com/share/cbfdd097a3aa4b029f3644a8310bc415

For this commit, I mainly added filter functionality for 4 tabs (all, today, this week, 1:1sessions) so that they could display corresponding course data. Also, when we open the detail modal, the apply button will take us to another confirmation page (this is still to be discussed).

And I added 'Add Courses' button on the sidebar, it will lead us to a page where instructors can add their own courses. This feature is still to be discussed.

### CODE REVIEW

**Review by ehsan2793 — APPROVED** [review]
Great job communicating with other frontend team members and using the dateConverter function to convert the date to something useful for the front end.
I have a question about the add course button you created. Is it going to be visible to all the users, or is it only visible to the instructor and admin? 
Overall fantastic job.

**Review by vannguyen141290 — APPROVED** [review]
glad that FE worked together to determined common ground before we built these functionalities. It helped our codes much more DRY and clean. I feel that we got much better code comparing with last week. Let's keep this good work until next week :)

File: src/components/pages/InstructorBooking/InstructorBookingCards.js [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
+  isOneToOneSession,
+} from '../../common/dateHelpers';
+
+const { TabPane } = Tabs;
 
 const InstructorBookingCards = props => {
   const { instructor } = props;
   const [currentModalProps, setCurrentModalProps] = useState(false);
   const [modalHidden, setModalHidden] = useState(true);
-  console.log('InstructorBookingCards', modalHidden);
```
Comment (TraNequaFauntleroy) [inline src/components/pages/InstructorBooking/InstructorBookingCards.js:11]: Good job cleaning this console.log() out

**Review by TraNequaFauntleroy — DISMISSED** [review]
I love how you cleaned up the code. I see you deleted a lot of files. I would just clean u the linter and dependency issues. Great work.

**Comment by huiminzhang0111** [comment]
> Great job communicating with other frontend team members and using the dateConverter function to convert the date to something useful for the front end. I have a question about the add course button you created. Is it going to be visible to all the users, or is it only visible to the instructor and admin? Overall fantastic job.

Great point!!! The add course function should only visible to instructors!

**Pull request merged**